### ISSUE TITLE
Make django-components look for templates in a specific place

### ISSUE THREAD

**Issue by oaltun** [EmilStenstrom/django-components#80]
Hi,

I just followed the documentation and installed the project.

I am trying to make the calendar component example work in an app called `tmp`. This is a cookiecutterdjangoproject, so apps are in a folder called `mysite`.

I am getting TemplateDoesNotExist error. Reason: my template is in `C:\Users\...\projectroot\mysite\tmp\components\calendar`, but django is searching `C:\Users\...\projectroot\mysite\templates` and `C:\Users\...\projectroot\mysite\tmp\templates`.

How should I solve this?

projectroot/mysite/tmp/components.py:

```
from django_components import component

@component.register("calendar")
class Calendar(component.Component):
    def context(self, date):
        return {
            "date": date,
        }

    def template(self, context):
        return "mysite/tmp/components/calendar/calendar.html"

    class Media:
        css = 'mysite/tmp/components/calendar/calendar.css'
        js = 'mysite/tmp/components/calendar/calendar.js'


```

Template settings:
```
TEMPLATES = [
    {
       
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [str(APPS_DIR / 'templates')], # APPS_DIR is projectroot/mysite
        'OPTIONS': {
            # https://docs.djangoproject.com/en/dev/ref/settings/#template-loaders
            # https://docs.djangoproject.com/en/dev/ref/templates/api/#loader-types
            'loaders': [
                'django.template.loaders.filesystem.Loader',
                'django.template.loaders.app_directories.Loader',
            ],
           ...

```

**Comment by EmilStenstrom** [comment]
Hi! What happens if you add the calendars directory to the DIRS setting? Does that solve the TemplateDoesNotExist error?

**Comment by oaltun** [comment]
It did work with the following arrangement.

files:
tmp/components.py
tmp/components/calendar/static/calendar/script.js
tmp/components/calendar/static/calendar/style.css
tmp/components/calendar/templates/calendar.html

components.py

```
from django_components import component

@component.register("calendar")
class Calendar(component.Component):
    def context(self, date):
        return {
            "date": date,
        }

    def template(self, context):
        return "calendar/calendar.html"

    class Media:
        css = 'calendar/style.css'
        js = 'calendar/script.js'
```

templates setting:

```
TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [
            str(APPS_DIR / 'templates'),
            str(APPS_DIR / 'tmp' / 'components' / 'calendar' / 'templates')
        ], # APPS_DIR is ../mysite

```

stitic files dirs setting:
```
STATICFILES_DIRS = [
    str(APPS_DIR / 'static'),
    str(APPS_DIR / 'tmp' / 'components' / 'calendar' / 'static'),
]
```

I guess somebody needs to write a template loader and a static files finder for django to automatically find these files.

**Comment by EmilStenstrom** [comment]
Thanks for posting the solution!

No matter what, you somehow need to specify the path to your templates. Are you suggesting we should come up with a default location for components, and make a loader that automatically loads templates from there?

**Comment by oaltun** [comment]
Right, default locations like <app_name>/components/<component_name>/templates for templates and <app_name>/components/<component_name>/static for static files would be nice I think.

**Comment by Real-Gecko** [comment]
I've encountered the same issue. I'm trying to add component to app named `common`. Here's how `common/components.py` looks like:
```python
from django_components import component

@component.register('permission_button')
class PermissionButton(component.Component):
    def context(self, permission, url, caption):
        return {
            'permission': permission,
            'url': url,
            'caption': caption,
        }

    def template(self, context):
        return 'common/components/permission_button/permission_button.html'
```
I get `TemplateDoesNotExist` unless template itself is not in `<BASE_DIR>/common/templates/common/components/permission_button/permission_button.html`

**Comment by Real-Gecko** [comment]
Never done this before, will investigate Django documentation on custom template loaders.

**Comment by Real-Gecko** [comment]
It turned out to be pretty easy, check #85 for details.

**Comment by EmilStenstrom** [comment]
Thanks to hard work of @Real-Gecko (THANKS!) this is no released as part of 0.17.
